services/kafka/service_test.go

```diff
@@ -33,7 +32,10 @@ func TestWriter_UpdateConfig(t *testing.T) {
 			PartitionAlgorithm: "",
 		}, []byte{1, 2, 3, 4}, []byte{1, 2, 3, 4}))
 
-	// Update the config in a way that requires shutting down all active writers.
 	c.UseSSL = true
 	require.NoError(t, cluster.Update(c))
+	diag.Close()
```

Comment:
Can any of the `Close` calls fail?  If so, let's `require.NoError` from them.

Reply:
Only one of those can return an error, but good idea.